### PR TITLE
fix(ivy): support module.id as @NgModule's "id" field value

### PR DESCRIPTION
**Pull request by AndrewKushnir**
Prior to this commit, the check that verifies correct "id" field type was too strict and didn't allow `module.id` as @NgModule's "id" field value. This change adds a special handling for `module.id` and uses it as @NgModule id as is. [Example of `module.id` provided by Webpack](https://webpack.js.org/api/module-variables/#moduleid-commonjs).

This PR resolves FW-1281.

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by kara — APPROVED** [review]
LGTM

**Comment by kara** [comment]
[presubmit](http://test/OCL:244780199:BASE:244780206:1555986048269:1a354b4b)

File: packages/compiler-cli/src/ngtsc/annotations/src/ng_module.ts [hunk 1]
```diff
@@ -121,15 +121,27 @@ export class NgModuleDecoratorHandler implements DecoratorHandler<NgModuleAnalys
       bootstrapRefs = this.resolveTypeList(expr, bootstrapMeta, name, 'bootstrap');
     }
 
-    let id: string|null = null;
+    let id: Expression|string|null = null;
```
Comment (alxhub) [inline packages/compiler-cli/src/ngtsc/annotations/src/ng_module.ts]: You know what? I don't think there's any point in validating that the `NgModule.id` is a string. Let's just wrap it with `WrappedNodeExpr` always.

**Review by alxhub — CHANGES_REQUESTED** [review]

File: packages/compiler-cli/src/ngtsc/annotations/src/ng_module.ts (hunk 1 above)
Comment (AndrewKushnir) [reply packages/compiler-cli/src/ngtsc/annotations/src/ng_module.ts]: Sounds great, that also simplifies the code a lot 👍

**Comment by AndrewKushnir** [comment]
@alxhub thanks for the review. I've updated this PR based on your proposal. Could you please have another look? Thank you.

**Review by alxhub — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
[Presubmit #2](http://test/OCL:244895316:BASE:244895337:1556045110525:68e406d)

**Pull request closed without merging**